manager/orchestrator/update/updater.go

```diff
@@ -520,10 +532,45 @@ func (u *Updater) removeOldTasks(ctx context.Context, batch *store.Batch, remove
 	return removedTask, nil
 }
 
+// Checks if the current assigned node matches the Placement.Constraints
+// specified in the task spec for Updater.newService.
+func (u *Updater) nodeMatches(t *api.Task) bool {
+	if t.Spec.Placement == nil {
+		return true
+	}
+
+	constraints, _ := constraint.Parse(u.newService.Spec.Task.Placement.Constraints)
+
+	match := false
+	u.store.Update(func(tx store.Tx) error {
+		n := store.GetNode(tx, t.NodeID)
+		match = constraint.NodeMatches(constraints, n)
+		return nil
+	})
+
+	return match
+}
+
+// Update the task spec for the given task with the one
+// specified in Updater.newService
+func (u *Updater) updateTaskServiceSpec(t *api.Task) error {
```

Comment:
Task specs are immutable. The orchestrator shouldn't update them.

Reply:
Just saw this comment. If this is the assumption then yes, we shouldn't be touching the task. As a side comment, it would make sense to enforce the immutability @aaronlehmann